### DISCUSSION TITLE
Non-existent aliases should default to error code 550 (hard bounce)

### DISCUSSION THREAD

**Discussion by junxit** [forwardemail/forwardemail.net#289] [Ideas]
I noticed today that if someone emails `non-existent-alias@example.com`, it does not bounce (goes to the blackhole).

This is less than ideal, since I as the owner/admin of `example.com` cannot always foretell what aliases people will use that I have not created yet.

As a workaround, I created `bounce@example.com` which is inactive and set to return error 550, then I created a catch-all `*@example.com` that forwards everything to `bounce@example.com`, and that works fine (as I expect it to).

Also, if I have a valid email alias `valid-alias@example.com` that forwards to `some-valid-account@gmail.com`, it still forwards correctly (no bouce) despite the `catch-all` that I set up.

Question posted by this chat thread:
1. Should the default behavior for non-existent aliases be to return SMTP error 550 (hard bounce)?
2. Should this behavior be made configurable at the domain level?
3. Alternatively, the catch-all (*) alias can be modified to allow it being added as `disabled` (today, disabling the `catch-all` forces you to remove it altogether), to achieve the behavior I get with my above workaround, without having to create a separate `bounce@example.com` that is disabled

Thoughts?

**Comment by shaunwarman** [comment]
Hey @thejade 

You could (and may already have based on the above) setup a catch all alias and then set the bounce reject rules appropriately. You're right it is a 250 quiet bounce (blackhole) by default today. But you could set a soft or hard reject.

<img width="927" alt="Screenshot 2024-09-09 at 8 29 59 AM" src="https://github.com/user-attachments/assets/3135e66e-f1f2-49fc-87b8-1cdbfcc0061e">

**Reply by junxit** [reply]
Unfortunately, this does not work. The settings you shared a screenshot for are only active if the `Active` checkbox is unchecked, and if you uncheck the box, you get this upon attempting to save the settings:
![image](https://github.com/user-attachments/assets/8dd5a255-7ce9-4351-8044-dcbb33a6477a)

This is why I suggested # 3 in my original post.

**Reply by shaunwarman** [reply]
Good call out. After some discussion we can default to 550 hard bounce. Just wanted to see if any issues around spam or security here. After that we can take a look at some extra configurability options based on your original message.

**Reply by junxit** [reply]
> Good call out. After some discussion we can default to 550 hard bounce. Just wanted to see if any issues around spam or security here. After that we can take a look at some extra configurability options based on your original message.

Agreed that someone can use the 550 or lack thereof to figure out if an alias exists or not. Making it configurable would likely be best.

**Comment by titanism** [comment]
By default all aliases that don't exist will reject with a standard `550` error code (as opposed to the `421` retry error code which we used since 2017).  We changed this behavior to match the defaults of major providers such as Gmail (which will hard reject with `550` if an alias doesn't exist).  You don't need to do anything, this already goes into effect.  Note that you still can customize a `250`, `421`, or `550` status code on a per-alias basis on both the free and paid plans already as-is.  This is just a general catch-all in case an alias doesn't exist (e.g. and you don't have a wild-card/catch-all setup).

We are deploying this now 🚀 

Ref: https://github.com/forwardemail/forwardemail.net/commit/592f30e5943d7a81deff8d25112ae76eeef1686f
